### ISSUE TITLE
Only the first did-you-mean-colon error in an object literal is reported

### ISSUE THREAD

**Issue by sandersn** [microsoft/TypeScript#47056]
```ts
const o = {
  x = 1,
  y = 2,
  #z: 3
}
```

Expected: 3 errors: Two did-you-mean-colon-not-equals error and one no-private-outside-class error.
Actual: 1 error: Just the first did-you-mean error.

The problem is that the check in `checkGrammarObjectLiteralExpression` concludes with `return grammarErrorOnNode`, whereas all the other checks don't return. The first step is to remove the `return`, and then make sure no other bogus errors get issued in the loop body that was previously getting skipped.
